python/ray/_private/utils.py

```diff
@@ -1191,6 +1191,42 @@ def check_dashboard_dependencies_installed() -> bool:
         return False
 
 
+def internal_kv_list_with_retry(gcs_client, prefix, namespace, num_retries=20):
```

Comment:
Do we need retries? Seems reasonable to just fail in this case presuming this is rare.